internal/inspect/inspect.go

```diff
@@ -1,104 +1,123 @@
 package inspect
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
+	"os"
 	"sort"
 	"strings"
 	"text/tabwriter"
 
 	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/app/internal"
 	"github.com/docker/app/render"
 	"github.com/docker/app/types"
+	"github.com/docker/app/types/metadata"
 	"github.com/docker/app/types/parameters"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	units "github.com/docker/go-units"
 )
 
+type service struct {
+	Name     string
+	Image    string
+	Replicas int
+	Mode     string
+	Ports    string
+}
+
+type attachment struct {
+	Path string
+	Size int64
+}
+
+type appInfo struct {
+	Metadata       metadata.AppMetadata
+	Services       []service
+	Networks       []string
+	Volumes        []string
+	Secrets        []string
+	parametersKeys []string
+	Parameters     map[string]string
+	Attachments    []attachment
+}
+
 // Inspect dumps the metadata of an app
 func Inspect(out io.Writer, app *types.App, argParameters map[string]string, imageMap map[string]bundle.Image) error {
+	outputFormat := os.Getenv(internal.DockerInspectFormatEnvVar)
+
 	// Render the compose file
 	config, err := render.Render(app, argParameters, imageMap)
 	if err != nil {
 		return err
 	}
 
-	// Extract all the parameters
-	parametersKeys, allParameters, err := extractParameters(app, argParameters)
+	appInfo, err := getAppInfo(app, config, argParameters)
 	if err != nil {
 		return err
 	}
 
+	if outputFormat == "json" {
```

Comment:
Can you extract the 2 formats to their own sub-function and add a `switch case` here?
```go
switch outputFormat:
case "json":
    return printJson(out, appInfo)
case "pretty":
    return printPretty(out, appInfo)
default:
    return fmt.Errorf("unknown format %q", outputFormat)
```